File: src/main/java/team4/footwithme/team/service/response/TeamInfoResponse.java
package team4.footwithme.team.service.response;

import team4.footwithme.team.domain.Team;

import java.util.List;

//팀 정보
public record TeamInfoResponse(
        String name,
        String description,
        String location,
        int winCount,
        int loseCount,
        int drawCount,
        List<String> evaluation,
        int maleCount,
        int femaleCount
) {

    public static TeamInfoResponse of(Team team, List<String> evaluation, int maleCount, int femaleCount){
        return new TeamInfoResponse(
                team.getName(),
                team.getDescription(),
                team.getLocation(),
                team.getTotalRecord().getWinCount(),
                team.getTotalRecord().getLoseCount(),
                team.getTotalRecord().getDrawCount(),
                evaluation,
                maleCount,
                femaleCount
        );
    }

}

File: src/main/java/team4/footwithme/team/service/response/TeamInfoResponse.kt
package team4.footwithme.team.service.response

import team4.footwithme.team.domain.Team

//팀 정보
@JvmRecord
data class TeamInfoResponse(
    @JvmField val name: String?,
    val description: String?,
    val location: String?,
    @JvmField val winCount: Int,
    val loseCount: Int,
    val drawCount: Int,
    @JvmField val evaluation: List<String?>,
    @JvmField val maleCount: Long?,
    @JvmField val femaleCount: Long?
) {
    companion object {
        fun of(team: Team, evaluation: List<String?>, maleCount: Long?, femaleCount: Long?): TeamInfoResponse {
            return TeamInfoResponse(
                team.name,
                team.description,
                team.location,
                team.totalRecord!!.winCount,
                team.totalRecord!!.loseCount,
                team.totalRecord!!.drawCount,
                evaluation,
                maleCount,
                femaleCount
            )
        }
    }
}
